pandas/core/indexes/multi.py

```diff
@@ -2715,7 +2716,7 @@ def difference(self, other):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
-        if not is_object_dtype(np.dtype(dtype)):
+        if not is_object_dtype(pandas_dtype(dtype)):
             raise TypeError('Setting %s dtype to anything other than object '
                             'is not supported' % self.__class__)
```

Comment:
Made this change since `np.dtype` doesn't recognize `'category'`, and would raise a different error message than the one specified here; `pandas_dtype` ensures that this error message will be raised.

Reply:
ahh yes, we should have very very limited ``np.dtype`` calls anywhere; ``pandas_dtype`` is the general version